gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveLocationDescriptor.java

```diff
@@ -52,14 +55,16 @@
   protected final FileSystem fileSystem;
   protected final Properties properties;
 
-  public Set<Path> getPaths() throws IOException {
+  public Map<Path, FileStatus> getPaths() throws IOException {
 
     PathFindingMethod pathFindingMethod = PathFindingMethod.valueOf(
         this.properties.getProperty(HIVE_LOCATION_LISTING_METHOD, DEFAULT_HIVE_LOCATION_LISTING_METHOD).toUpperCase());
 
+    Map<Path, FileStatus> result = Maps.newHashMap();
```

Comment:
I am concerned with the performance hit that a `BiMap` might introduce without clear benefit. Discussed offline.